lib/samples/download_preplanned_map_area/download_preplanned_map_area_sample.dart

```diff
@@ -34,130 +32,275 @@ class DownloadPreplannedMapAreaSample extends StatefulWidget {
-              height: 235,
-              width: double.infinity,
-              child: _buildMapAreaList(context),
-            )
           ],
         ),
       ),
+      // Display a list of available maps in a bottom sheet.
+      bottomSheet:
+          _mapSelectionVisible ? buildMapSelectionSheet(context) : null,
     );
   }
 
-  ListView _buildMapAreaList(BuildContext context) {
-    return ListView.builder(
-      padding: const EdgeInsets.all(8),
-      itemCount: _preplannedAreas.length,
-      itemBuilder: (context, index) {
-        final name = _preplannedAreas[index].portalItem.title;
-        return Card(
-          child: ListTile(
-            trailing: Text((index == _selectedPreplannedAreaIndex && _isLoading)
-                ? 'Loading'
-                : ''),
-            selected: (index == _selectedPreplannedAreaIndex),
-            title: Text(name),
-            onTap: () async {
-              if (index == _selectedPreplannedAreaIndex) {
-                return;
-              }
-              setState(() {
-                _selectedPreplannedAreaIndex = index;
-                _isLoading = true;
-              });
-              try {
-                final map = await _downloadOfflineMap(index);
-                _mapViewController.arcGISMap = map;
-              } catch (e) {
-                print(e);
-              } finally {
-                setState(() => _isLoading = false);
-              }
+  void onMapViewReady() async {
+    // Configure the directory to download offline maps to.
+    _downloadDirectory = await createDownloadDirectory();
+
+    // Create a map using a webmap from a portal item.
+    final portal = Portal.arcGISOnline();
+    final portalItem = PortalItem.withPortalAndItemId(
+        portal: portal, itemId: 'acc027394bc84c2fb04d1ed317aac674');
+    _webMap = ArcGISMap.withItem(portalItem);
+
+    // Create and load an offline map task using the portal item.
+    _offlineMapTask = OfflineMapTask.withPortalItem(portalItem);
+    await _offlineMapTask.load();
+
+    // Get the preplanned map areas from the offline map task and load each.
+    final preplannedMapAreas = await _offlineMapTask.getPreplannedMapAreas();
+    for (final mapArea in preplannedMapAreas) {
+      await mapArea.load();
+      // Add each map area as a key in the Map, setting the Job to null initially.
+      _preplannedMapAreas[mapArea] = null;
+    }
+
+    // Initially set the web map to the map view controller.
+    _mapViewController.arcGISMap = _webMap;
+
+    // Upate the state with the preplanned map areas and set the UI state to ready.
+    setState(() => _ready = true);
+  }
+
+  // Builds a map selection widget with a list of available maps.
+  Widget buildMapSelectionSheet(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.fromLTRB(
+          20.0,
+          20.0,
+          20.0,
+          View.of(context).viewPadding.bottom /
+              View.of(context).devicePixelRatio),
```

Comment:
To accommodate devices with no SafeArea on the bottom, we later figured out the bottom formula needs to be:
```dart
max(
  20.0,
  View.of(context).viewPadding.bottom /
    View.of(context).devicePixelRatio,
),
```
(and import `dart:math`).

This gives us a minimum of 20.0 bottom padding, which only kicks in if there's no system widgets taking up space on the bottom (e.g., iPhone SE).